lib/SDCard/SDCard.cpp

```diff
@@ -4,85 +4,103 @@
 
 #include <definitions.h>
 
+// standard libraries
+#include <fmt/core.h>
+#include <iostream>
+#include <stdio.h>
+#include <string>
+
+#include <FS.h>
 #include <SD.h> // sd card
+#include <SD_MMC.h>
 #include <SPI.h>
 
 #include <DriverDisplay.h>
 #include <Helper.h>
 #include <SDCard.h>
 #include <SPIBus.h>
 
-#define FILENAME "/test.txt"
+#define FILENAME "/ser4data.log"
 
-extern SPIClass SPI;
 extern SPIBus spiBus;
 extern SDCard sdCard;
 extern DriverDisplay driverDisplay;
 
 void SDCard::re_init() { init(); }
 
+void printDirectory(File dir, int numTabs) {
+  while (true) {
+
+    File entry = dir.openNextFile();
+    if (!entry) {
+      // no more files
+      break;
+    }
+    for (uint8_t i = 0; i < numTabs; i++) {
+      Serial.print('\t');
+    }
+    Serial.print(entry.name());
+    if (entry.isDirectory()) {
+      Serial.println("/");
+      printDirectory(entry, numTabs + 1);
+    } else {
+      // files have sizes, directories do not
+      Serial.print("\t\t");
+      Serial.println(entry.size(), DEC);
```

Comment:
Everywhere else you changed to Cpp-Style `cout << `

Reply:
done,